src/App.tsx

```diff
@@ -1,72 +1,107 @@
-import React from 'react';
 import './App.scss';
+import React, { useState, useCallback } from 'react';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import debounce from 'lodash.debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [searchPart, setSearchPart] = useState('');
+  const [suggestedPeople, setSuggestedPeople] = useState(peopleFromServer);
+  const [currentPerson, setCurrentPerson] = useState<Person | null>(null);
+  const [isDisplayed, setIsDisplayed] = useState(false);
+
+  const filterPeople = (value: string) => {
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(value.toLowerCase()),
+    );
+  };
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedSetSearchPart = useCallback(
+    debounce((value: string) => {
+      setSuggestedPeople(filterPeople(value));
+      setIsDisplayed(true);
+    }, 300),
+    [],
+  );
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value.trim();
+
+    setSearchPart(value);
+    setCurrentPerson(null);
+    debouncedSetSearchPart(value);
+  };
+
+  const handleChoose = (person: Person) => {
+    setCurrentPerson(person);
+    setSearchPart(person.name);
+    setIsDisplayed(false);
+
+    setTimeout(() => {
+      setSuggestedPeople(filterPeople(''));
+      setIsDisplayed(false);
+    }, 0);
```

Comment:
```suggestion
    setIsDisplayed(false);
```
You definitely shouldn't do it `setIsDisplayed(false)` twice
I guess you can just delete `setTimeout`. In case it doesn't work anymore just leave timeout and do `setIsDisplayed` once inside timeout

setSuggestedPeople(filterPeople('')); seems redundant to me here - next time you will show list of people you will redefine it anyway